app/controllers/users/sessions_controller.rb

```diff
@@ -1,3 +1,13 @@
 # frozen_string_literal: true
 
-class Users::SessionsController < Devise::SessionsController; end
+class Users::SessionsController < Devise::SessionsController
+  def create
+    self.resource = warden.authenticate!(auth_options)
```

Comment:
Just curious why we are overriding devise here.

Reply:
Because Devise believes in a different way of surfacing errors than we do.

Devise likes to have global `errors`, which it wants to surface as a notice. 
We like to have errors on models that we made form with - so we can automatically link them to the right fields.
And unfortunately, devise doesn't put those errors on the `resource` :/ 

And this is the simplest hack I found to get the behaviour we want - to add an error to the `email` fiend on the user.